logstash-core/lib/logstash/runner.rb

```diff
@@ -296,6 +296,10 @@ def execute
     # override log level that may have been introduced from a custom log4j config file
     LogStash::Logging::Logger::configure_logging(setting("log.level"))
 
+    if log_configuration_contains_javascript_usage?
+      logger.warn("Logging configuration uses Script log appender or filter with Javascript, which will be removed in a future major release of Logstash.")
```

Comment:
nit: would have been nice if this used `JavaScript` name - since that is the string we detect in the configuration